bin/node/src/config.rs

```diff
@@ -1,54 +1,92 @@
 use miden_node_block_producer::config::BlockProducerConfig;
 use miden_node_rpc::config::RpcConfig;
 use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::Endpoint;
 use serde::{Deserialize, Serialize};
 
 /// Node top-level configuration.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+#[derive(Clone, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
 pub struct NodeConfig {
-    pub block_producer: Option<BlockProducerConfig>,
-    pub rpc: Option<RpcConfig>,
-    pub store: Option<StoreConfig>,
+    block_producer: NormalizedBlockProducerConfig,
+    rpc: NormalizedRpcConfig,
+    store: StoreConfig,
+}
+
+/// A specialized variant of [RpcConfig] with redundant fields within [NodeConfig] removed.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+struct NormalizedRpcConfig {
```

Comment:
Should we also add `#[serde(deny_unknown_fields)]` here and to other config structs in this module?

Reply:
Yes I somehow missed this, thanks.